pyblish/util.py

```diff
@@ -73,16 +125,42 @@ def publish(context=None, plugins=None, targets=None):
     }
 
     # Second pass, the remainder
+    plugin_processing = None
+    instances_processed_count = 0
+    total_plugins = len(plugins) + len(collectors)
     for Plugin, instance in logic.Iterator(plugins, context, state):
         try:
             result = plugin.process(Plugin, context, instance)
 
+            # Calculate percentage
+            instances_count = len(logic.instances_by_plugin(context, Plugin))
```

Comment:
Do we need to be this exact...? Maybe.. You could also make `logic.Iterator` into a list, which would give you a fixed series of tasks to perform, and get the progress from `task_number / len(tasks)`

Reply:
You are right, just tested it and it works better plus simplifies the code a lot. Pushed the code for a review.